controller/controller_client.go

```diff
@@ -23,266 +20,234 @@ type doneChan chan struct{}
 
 // A little helper for managing the state of the main runloop
 type clientHandler struct {
-	// tells the shipper server to begin shutdown
-	shutdownInit doneChan
-	// Tells the controller that the shipper backend has gracefully shut down.
-	shutdownComplete sync.WaitGroup
-
-	unitsMut sync.Mutex
-	units    map[string]*client.Unit
-
-	// the unit ID for the shipper itself
-	shipperOutputID string
+	// hashmap of units we get
+	units *UnitMap
 
-	shipperIsStopping uint32
-
-	log *logp.Logger
+	//shipperIsStopping uint32
+	shipperIsRunning bool
+	runner           *ServerRunner
+	log              *logp.Logger
 }
 
 func newClientHandler() clientHandler {
 	return clientHandler{
-		shutdownInit:      make(doneChan, 1),
-		log:               logp.L(),
-		units:             make(map[string]*client.Unit),
-		shipperIsStopping: 0,
+		log:              logp.L(),
+		units:            NewUnitMap(),
+		shipperIsRunning: false,
 	}
 }
 
-// Run starts the gRPC server
-func (c *clientHandler) Run(cfg config.ShipperConfig, unit *client.Unit) (err error) {
-	_ = unit.UpdateState(client.UnitStateConfiguring, "Initialising shipper server", nil)
-	runner, err := NewServerRunner(cfg)
-	if err != nil {
-		return err
+/*
+//////////
+////////// Shipper/output Handlers
+/////////
+*/
+
+// BounceShipper performs a (re)start of the shipper's backend components when triggered by another component
+func (c *clientHandler) BounceShipper() {
```

Comment:
Bounce is not the clearest name for this, why not something more literal like Restart or Reload or something?